### PR TITLE
[Feature][E2E] add e2e container for flink 1.17 1.18

### PR DESCRIPTION
**Pull request by TyrantLucifer**
### Purpose of this pull request

Add flink 1.17 1.18 e2e test cases.

### How was this patch tested?

Existed test cases.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink18Container.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    protected String getStartModuleName() {
+        return "seatunnel-flink-starter" + File.separator + "seatunnel-flink-15-starter";
+    }
+
+    @Override
+    protected String getStartShellName() {
+        return "start-seatunnel-flink-15-connector-v2.sh";
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink18Container.java:54]: shall we change shell script name?

**Review by Hisoka-X — DISMISSED** [review]
LGTM

**Comment by TyrantLucifer** [comment]
> LGTM

CI seems that has some problems. Give me some time to resolve it

**Comment by hailin0** [comment]
LGTM

waiting for ci passed

File: seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink17Container.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @Override
+    protected String getStartModuleName() {
+        return "seatunnel-flink-starter" + File.separator + "seatunnel-flink-15-starter";
+    }
+
+    @Override
+    protected String getStartShellName() {
+        return "start-seatunnel-flink-15-connector-v2.sh";
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink17Container.java:54]: Same as above

File: seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink18Container.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink18Container.java:54]: I think we do not need to change it, from flink 1.15 to flink 1.18，core api doesn't be changed.

Comment (TyrantLucifer) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink18Container.java:54]: No need, as the API is compatible

File: seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink17Container.java (hunk 2 above)
Comment (TyrantLucifer) [reply seatunnel-e2e/seatunnel-e2e-common/src/test/java/org/apache/seatunnel/e2e/common/container/flink/Flink17Container.java:54]: No need, as the API is compatible

**Review by Hisoka-X — DISMISSED** [review]

**Review by hailin0 — DISMISSED** [review]

**Review by TaoZex — DISMISSED** [review]
+1

**Pull request closed without merging**